CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/HabitFragments.java

```diff
@@ -0,0 +1,194 @@
+                else
+                    calendarView.setVisibility(View.GONE);
+                //calendarView.setVisibility(calendarView.getVisibility() == View.GONE ? View.VISIBLE : View.GONE);
+            }
+        });
+
+        Bundle args = getArguments();
+        if (args != null) {
+            Habit Retrieved_Habit = (Habit)args.getSerializable("habit");
+            habitTitle.setText(Retrieved_Habit.getHabitName());
+            startDate.setText(Retrieved_Habit.getReason());
+            AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+            return builder
+                    .setView(view)
+                    .setTitle("Add Habit")
+                    .setNegativeButton("Cancel", null)
+                    .setPositiveButton("OK", new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialog, int i) {
+
+                            String name = habitTitle.getText().toString();
+                            String reason = startDate.getText().toString();
+                            Retrieved_Habit.setHabitName(habitTitle.getText().toString());
+                            Date date = new Date(calendarView.getDate());
+                            byte weekdays = 0;
+                            CheckBox checkBox = view.findViewById(R.id.mondayBox);
+                            if (checkBox.isChecked() == true)
+                                weekdays = (byte) (weekdays + 1);
+                            checkBox = view.findViewById(R.id.tuesdayBox);
+                            if (checkBox.isChecked() == true)
+                                weekdays = (byte) (weekdays + 2);
+                            checkBox = view.findViewById(R.id.wednesdayBox);
+                            if (checkBox.isChecked() == true)
+                                weekdays = (byte) (weekdays + 4);
+                            checkBox = view.findViewById(R.id.thursdayBox);
+                            if (checkBox.isChecked() == true)
+                                weekdays = (byte) (weekdays + 8);
+                            checkBox = view.findViewById(R.id.fridayBox);
+                            if (checkBox.isChecked() == true)
+                                weekdays = (byte) (weekdays + 16);
+                            checkBox = view.findViewById(R.id.saturdayBox);
+                            if (checkBox.isChecked() == true)
+                                weekdays = (byte) (weekdays + 32);
+                            checkBox = view.findViewById(R.id.sundayBox);
+                            if (checkBox.isChecked() == true)
+                                weekdays = (byte) (weekdays + 64);
+                            listener.onOkPressed(Retrieved_Habit, name,reason, date, weekdays);
+                        }
+                    }).create();
+        }
+        else {
```

Comment:
This chunk of code under here seems identical to the code above starting from line 103. I don't think there's a reason to lock it into a conditional then. Perhaps just put what's different (lines 100-102) inside the conditional and only have one instance of this giant chunk of code.